### DISCUSSION TITLE
Want to contribute to the Project but unable to setup IDE.

### DISCUSSION THREAD

**Discussion by sanchit-sign3** [apache/pulsar#19725] [Q&A]
Hi folks,

I wish to contribute to the pulsar but can't setup intellij.
I followed the instructions mentioned in https://pulsar.apache.org/contribute/setup-ide/ but still came short.

Can someone please help. Perhaps 15mins of their precious time.

**Comment by jak78** [comment]
Hi @sanchit-sign3, and welcome to the community! I'd be happy to help. Could you please describe the issue you are facing?

**Reply by sanchit-sign3** [reply]
So I am using **intellij**. I configured the project to use jdk17 and maven to use the project JDK. I am able to build the project using "mvn clean install" via the command line.
Problem: 
1. In each module, the IDE  is able to recognise the files as java files but not as java classes. Basically the IDE is not picking the individual java folders of the modules as "Sources root". 
2. Also some of the poms have **relativePath** tag whose value being ".." is not being resolved.
<img width="1512" alt="Screenshot 2023-03-06 at 9 12 45 PM" src="https://user-images.githubusercontent.com/98951764/223158898-9e897f46-6a41-41b6-b845-67eaf995b282.png">
<img width="1512" alt="Screenshot 2023-03-06 at 9 12 59 PM" src="https://user-images.githubusercontent.com/98951764/223158950-24477b00-ad21-46e4-aae2-e00c04cb1bec.png">

**Comment by tisonkun** [comment]
We have a setting up IDE guide https://pulsar.apache.org/contribute/setup-ide.

For your concrete case, the relativePath warning is a known issue, should not affect indexing and code intelligence.

For loading maven modules, you may reload maven modules from right click menuum in any pom.xml file. If the issue remains, try invalidate cache.

This seems common IDEA issues, not caused by Pulsar code base. And I can configure the Pulsar project with JDK 17 in IDEA so there should not be steady blockers.

**Comment by sanchit-sign3** [comment]
So I did the following as well:
1. Already reloaded maven modules from right click menu in any pom.xml file
2. Invalidated the cache
<img width="1512" alt="Screenshot 2023-03-06 at 9 57 47 PM" src="https://user-images.githubusercontent.com/98951764/223170980-ae493f5a-4e46-4f93-8a85-05abd8945990.png">

I also tried marking the java folder of say "pulsar-broker-common" module as "Sources root", And now the jdk is recognized by the files but still the annotations are unrecognized even though I followed the annotation settings instructions in the https://pulsar.apache.org/contribute/setup-ide.

**Reply by jak78** [reply]
Could you double-check you have installed the Lombok plugin?
Did you also try to rebuild the project? Build -> Rebuild Project

**Comment by sanchit-sign3** [comment]
Yep. I did it all.
And the screenshot where you see that lombok annotations(ex - @Slf4j) are
not acknowledged, I explicitly marked the java folder as sources root.
It still doesn't work. Do you wanna get on a quick zoom call perhaps? I
would really appreciate it.
